### PR TITLE
Finding IP Address of a URL in Java (biivincent4@gmail.com) 

### PR DESCRIPTION
**Pull request by vincentbii**

### CODE REVIEW

File: core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Country.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    private String name;
+    private Currency currency;
+
+    public Country(String code, String name, Currency currency) {
+        this.code = code;
+        this.name = name;
+        this.currency = currency;
+    }
+
+    public Country(Country that) {
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Country.java:18]: maybe let's replace _that_ by _country_

_that_ can be a bit confusing

File: core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Currency.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    private String symbol;
+    private BigDecimal value;
+
+    public Currency(String code, String symbol, BigDecimal value) {
+        this.code = code;
+        this.symbol = symbol;
+        this.value = value;
+    }
+
+    public Currency(Currency that) {
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Currency.java:18]: same here

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    }
+
+    public String getBySocketConnection(String urlString) throws IOException {
+        try (Socket socket = new Socket()) {
+            socket.connect(new InetSocketAddress(urlString, 80));
+            return socket.getLocalAddress().getHostAddress();
+        }
+    }
+
```
Comment (vinipx) [inline core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java:25]: you can remove these blank lines

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import java.net.UnknownHostException;
+
+public class URLIPAddress {
+
+    public String getByInetAddress(String urlString) throws UnknownHostException {
+
+        InetAddress ip = InetAddress.getByName(urlString);
+
+        return ip.getHostAddress();
+
```
Comment (vinipx) [inline core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java:16]: let's remove this blank line

**Review by vinipx — CHANGES_REQUESTED** [review]

**Comment by vincentbii** [comment]
update with the changes requested

File: core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Country.java (hunk 1 above)
Comment (vincentbii) [reply core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Country.java]: check the latest commit. this is resolved

File: core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Currency.java (hunk 2 above)
Comment (vincentbii) [reply core-java-modules/core-java-lang-oop-patterns/src/main/java/com/baeldung/cloning/Currency.java]: proceed to review

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java (hunk 3 above)
Comment (vincentbii) [reply core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java]: proceed to review

File: core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java (hunk 4 above)
Comment (vincentbii) [reply core-java-modules/core-java-networking-3/src/main/java/com/baeldung/urlipaddress/URLIPAddress.java]: proceed to review

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/DeepCloningTest.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+public class DeepCloningTest {
+
+    @Test
+    public void whenCloningObject_thenObjectsShouldNotBeTheSame() throws CloneNotSupportedException {
+        Country country = new Country("KE", "Kenya", new Currency("KES", "Ksh", BigDecimal.valueOf(1000)));
+        Country deepClone = country.clone();
+
+        assertThat(country).isNotSameAs(deepClone);
+        assertThat(country.getCurrency()).isNotSameAs(deepClone.getCurrency());
+
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/DeepCloningTest.java:18]: minor, but lets remove this empty line here

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/ShallowCopyTest.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        Country shallowClone = new Country(country.getCode(), country.getName(), country.getCurrency());
+
+        country.getCurrency()
+            .setCode("KE");
+        assertThat(country).isNotSameAs(shallowClone);
+        assertThat(country.getCurrency()).isSameAs(shallowClone.getCurrency());
+        assertThat(country.getCurrency()
+            .getCode()).isEqualTo(shallowClone.getCurrency()
+            .getCode());
+
```
Comment (vinipx) [inline core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/ShallowCopyTest.java:33]: minor: do we need these empty lines here? maybe remove one at least

**Comment by vinipx** [comment]
@vincentbii you could use package private access modifier in test methods

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/ShallowCopyTest.java (hunk 6 above)
Comment (vincentbii) [reply core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/ShallowCopyTest.java]: resolved

File: core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/DeepCloningTest.java (hunk 5 above)
Comment (vincentbii) [reply core-java-modules/core-java-lang-oop-patterns/src/test/java/com/baeldung/cloning/DeepCloningTest.java]: resolved

**Comment by vinipx** [comment]
@vincentbii pls check the tests - you've set them as private and some are public. Ideally, as per guidelines, we should use **package private** in our Java code.

**Comment by vincentbii** [comment]
@vinipx 

i seem not to understand the problem. Can I get a sample

Going through other code, and test cases are using `public void ...` what I was using before

Please guide

**Comment by vinipx** [comment]
> @vinipx
> 
> i seem not to understand the problem. Can I get a sample
> 
> Going through other code, and test cases are using `public void ...` what I was using before
> 
> Please guide

Sure, let me clarify.

As per the guidelines, we have a recommendation to use package scope (package private) wherever possible:
>- use package-scope access modifier if possible

meaning that test methods should look like this (using default access modifier): 
`void givenXXX_whenXXX_thenXXX() {}`

**Comment by vincentbii** [comment]
would you confirm now,

i am using `void when...`

**Pull request merged**